src/full/Agda/Utils/Environment.hs

```diff
@@ -25,6 +28,13 @@ expandVars home env s = concatMap repl $ tokens s
     repl (Var x) = fromMaybe "" $ lookup x env
     repl (Str s) = s
 
+-- | Expand a telescope of environment variables
+--   (each value may refer to variables earlier in the list)
+expandEnvVarTelescope :: String -> [(String, String)] -> [(String, String)]
+expandEnvVarTelescope home vs0 = reverse $ foldr  -- compensate for reverse below
+  (\(var,val) acc -> (var, expandVars home acc val):acc)
+  [] (reverse vs0)  -- reverse because earlier vals must be processed first
```

Comment:
Isn't a `foldr . reverse` a `foldl`?

Reply:
I see two constraints:

* Expansion must happen left-to-right
* The output list must have the same order as the input list

This works: `reverse . foldr f [] . reverse`
This also works: `reverse . foldl (flip f) []`